pkg/server/registry/apigroups/acorn/images/detail.go

```diff
@@ -54,23 +58,42 @@ func (s *ImageDetailStrategy) Create(ctx context.Context, obj types.Object) (typ
 	}
 	ns, _ := request.NamespaceFrom(ctx)
 	opts := []remote.Option{s.remoteOpt}
+	imageName := strings.ReplaceAll(details.ImageName, "+", "/")
 	if details.Auth != nil {
-		imageName := strings.ReplaceAll(details.ImageName, "+", "/")
 		ref, err := name.ParseReference(imageName)
 		if err == nil {
 			opts = append(opts, remote.WithAuthFromKeychain(images.NewSimpleKeychain(ref.Context(), *details.Auth, nil)))
 		}
 	}
-	return imagedetails.GetImageDetails(ctx, s.client, ns, details.ImageName, imagedetails.GetImageDetailsOptions{
+	id, err := imagedetails.GetImageDetails(ctx, s.client, ns, details.ImageName, imagedetails.GetImageDetailsOptions{
 		Profiles:      details.Profiles,
 		DeployArgs:    details.DeployArgs,
 		Nested:        details.NestedDigest,
 		NoDefaultReg:  details.NoDefaultRegistry,
 		IncludeNested: details.IncludeNested,
 		RemoteOpts:    opts,
 	})
+
+	return id, translateRegistryErrors(err, imageName)
 }
 
 func (s *ImageDetailStrategy) New() types.Object {
 	return &apiv1.ImageDetails{}
 }
+
+func translateRegistryErrors(in error, imageName string) error {
+	if in == nil {
+		return nil
+	}
+	if terr, ok := in.(*transport.Error); ok {
+		switch terr.StatusCode {
+		case http.StatusNotFound:
+			return errors.NewNotFound(schema.GroupResource{Resource: "images"}, imageName)
+		case http.StatusUnauthorized:
+			return errors.NewUnauthorized(fmt.Sprintf("pulling image %s: %v", imageName, terr))
+		case http.StatusForbidden:
+			return errors.NewForbidden(schema.GroupResource{Resource: "images"}, imageName, terr)
```

Comment:
Simple change, but I'm not sure if that makes sense, since it's not necessarily an image resource that wasn't found but potentially an external image, where there's no API representation (yet).
Then again it also doesn't harm, so I'll add it :)